### PR TITLE
Bug 1832006 - [MozillaOnline] Add description of site permissions fragment

### PR DESCRIPTION
**Pull request by rxumoz**
Due to the requirement from authorities in China, there will be several PR about MozillaOnline builds modifications in the next few days.

This PR add description of site permissions fragment to make it clearer to users.

https://bugzilla.mozilla.org/show_bug.cgi?id=1832006
https://bugzilla.mozilla.org/show_bug.cgi?id=1832006
https://bugzilla.mozilla.org/show_bug.cgi?id=1832006
https://bugzilla.mozilla.org/show_bug.cgi?id=1832006
https://bugzilla.mozilla.org/show_bug.cgi?id=1832006

### CODE REVIEW

File: fenix/app/src/main/res/xml/site_permissions_preferences.xml [hunk 1]
```diff
@@ -2,9 +2,17 @@
 <!-- This Source Code Form is subject to the terms of the Mozilla Public
    - License, v. 2.0. If a copy of the MPL was not distributed with this
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
-<androidx.preference.PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android"
+<androidx.preference.PreferenceScreen xmlns:tools="http://schemas.android.com/tools"
+    xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto">
 
+    <androidx.preference.PreferenceCategory
```
Comment (rocketsroger) [inline fenix/app/src/main/res/xml/site_permissions_preferences.xml:9]: Can we please make this visibility default false and set it to visible.  See above

File: fenix/app/src/main/java/org/mozilla/fenix/settings/sitepermissions/SitePermissionsFragment.kt [hunk 2]
```diff
@@ -23,6 +24,11 @@ class SitePermissionsFragment : PreferenceFragmentCompat() {
 
     override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
         setPreferencesFromResource(R.xml.site_permissions_preferences, rootKey)
+
+        val preferenceDescription = requirePreference<Preference>(R.string.pref_key_site_permissions_description)
+        if (!Config.channel.isMozillaOnline) {
```
Comment (rocketsroger) [inline fenix/app/src/main/java/org/mozilla/fenix/settings/sitepermissions/SitePermissionsFragment.kt:29]: I think it'll be better if by default visibility is false and we check here to set to true.
```
if (Config.channel.isMozillaOnline) {
            preferenceDescription.isVisible = true
        }
```

**Review by rocketsroger — COMMENTED** [review]
Looks good

File: fenix/app/src/main/java/org/mozilla/fenix/settings/sitepermissions/SitePermissionsFragment.kt (hunk 2 above)
Comment (rxumoz) [reply fenix/app/src/main/java/org/mozilla/fenix/settings/sitepermissions/SitePermissionsFragment.kt]: Okay, I'll update.

**Review by rocketsroger — APPROVED** [review]

**Comment by rxumoz** [comment]
@gabrielluong Could you help to merge this? Since I'm not authorized.

**Comment by gabrielluong** [comment]
@rxumoz Do you have permissions to add labels? We just add the "needs landing" label to queue it up for merging.

**Comment by rxumoz** [comment]
> @rxumoz Do you have permissions to add labels? We just add the "needs landing" label to queue it up for merging.

No, I can't label...

**Pull request merged**